### PR TITLE
Add warning when retrieving state cache inefficiency detected

### PR DESCRIPTION
**Pull request by Abacn**
**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by Abacn** [comment]
example: 
image

File: sdks/python/apache_beam/runners/worker/sdk_worker.py [hunk 1]
```diff
@@ -1272,14 +1277,40 @@ def _lazy_iterator(
...
+    input_stream = coder_impl.create_InputStream(data)
+
+    self._retrieval_time += time.time() - start_time
+    self._get_raw_called += 1
+
+    if self._retrieval_time > self._warn_interval:
+      _LOGGER.warning(
+          f"Retrieving state {self._get_raw_called} times costed "
+          f"{self._retrieval_time:.0f} seconds. It may be due to insufficient "
+          "state cache size and/or frequent direct access of states.\n"
```
Comment (liferoad) [inline sdks/python/apache_beam/runners/worker/sdk_worker.py:1304]: possible to show the current cache size? and show the option to change it? basically make it more informative.

**Review by riteshghorse — APPROVED** [review]
LGTM mod lint failures. Thank you for this PR!

File: sdks/python/apache_beam/runners/worker/sdk_worker.py (hunk 1 above)
Comment (riteshghorse) [reply sdks/python/apache_beam/runners/worker/sdk_worker.py:1304]: I think current cache size could be fetched with `self._state_cache.size()`

Comment (Abacn) [reply sdks/python/apache_beam/runners/worker/sdk_worker.py:1304]: phrased to more specific message.

There is a INFO worker log like "Creating state cache with size 26214400" for every harness started

**Comment by Abacn** [comment]
what it looks like now:

image

**Review by liferoad — APPROVED** [review]

**Pull request merged**